src/main/scala/com/databricks/labs/overwatch/pipeline/InitializerFunctions.scala

```diff
@@ -536,4 +495,93 @@ trait InitializerFunctions
    */
   def validateAndSetDataTarget(dataTarget: DataTarget): Unit
 
+  def validateAuditLogConfigsFromSystemTable(auditLogConfig: AuditLogConfig,
+                                             organizationId: String,
+                                             workspace_url: String,
+                                             token: Option[TokenSecret]): AuditLogConfig = {
+    val auditLogFormat = "delta"
+    val systemTableName = auditLogConfig.systemTableName.get
+    val sqlEndpoint = auditLogConfig.sqlEndpoint.getOrElse("")
+    if(sqlEndpoint.isEmpty) {
+      val systemTableNameDf = spark.table(systemTableName).filter(s"workspace_id = '$organizationId'")
+      if (systemTableNameDf.isEmpty)
+        throw new Exception(s"No data found in ${systemTableName} for organizationId: $organizationId ")
+      auditLogConfig.copy(auditLogFormat=auditLogFormat,systemTableName = Some(systemTableName))
+    }
+    else {
+      val host = workspace_url.stripPrefix("https://").stripSuffix("/")
+      val scope = token.get.scope
+      val key = token.get.key
+      val rawToken = dbutils.secrets.get(scope, key)
+      val systemTableNameDf = spark.read
+        .format("databricks")
+        .option("host", host)
+        .option("httpPath", sqlEndpoint)
+        .option("personalAccessToken", rawToken)
+        .option("query", s"select * from ${systemTableName} where workspace_id='${organizationId}'")
```

Comment:
add limit 1 to reduce the amount of data to transfer, since we're just checking that the connection can happen

Reply:
done